src/Microsoft.DotNet.ImageBuilder/src/Commands/CleanAcrImagesCommand.cs

```diff
@@ -233,5 +248,24 @@ private async Task DeleteRepositoryAsync(IContainerRegistryClient acrClient, Lis
         }
 
         private static bool IsExpired(DateTimeOffset dateTime, int expirationDays) => dateTime.AddDays(expirationDays) < DateTimeOffset.Now;
+
+        private async Task<bool> IsAnnotationManifestAsync(ArtifactManifestProperties manifest, IContainerRegistryContentClient acrContentClient)
+        {
+            ManifestQueryResult manifestResult = acrContentClient.GetManifestAsync(manifest.Digest).Result;
+
+            // An annotation is just a referrer and referrers are indicated by the presence of a subject field.
+            return manifestResult.Manifest["subject"] is not null;
+        }
+
+        private async Task<bool> HasExpiredEolAsync(ArtifactManifestProperties manifest, int expirationDays)
+        {
+            if(_orasService.IsDigestAnnotatedForEol(manifest.RegistryLoginServer + "/" + manifest.RepositoryName + "@" + manifest.Digest, _loggerService, Options.IsDryRun, out OciManifest? lifecycleArtifactManifest) &&
```

Comment:
Explicitly set dry run to false here. The command has no side effects, so it's fine. It's necessary to do this because we want to allow using dry run for the clean command. But if we enable dry run for running the oras tool, the tool never gets invoked and this condition would never be true, giving incorrect results for the command to analyze.

```suggestion
            if(_orasService.IsDigestAnnotatedForEol(manifest.RegistryLoginServer + "/" + manifest.RepositoryName + "@" + manifest.Digest, _loggerService, isDryRun: false, out OciManifest? lifecycleArtifactManifest) &&
```